hadoop-tools/hadoop-aws/src/main/java/org/apache/hadoop/fs/s3a/impl/V2Migration.java

```diff
@@ -28,83 +28,47 @@
 /**
  * This class provides utility methods required for migrating S3A to AWS Java SDK V2.
  * For more information on the upgrade, see HADOOP-18073.
+ *
+ * <p>in HADOOP-18382. Upgrade AWS SDK to V2 - Prerequisites,
+ * this class contained a series of `LogExactlyOnce` loggers to warn on
+ * the first use of a feature which would change incompatibly; this shipped in Hadoop 3.3.5.
+ * <p>
+ * With the move to v2 completed, attempts to use the v1 classes, will fail
+ * -except for the special case of support for v1 credential providers.
+ * <p>
+ * The warning methods are still present, where appropriate, but downgraded to debug
+ * and only retained for debugging migration issues.
  */
 public final class V2Migration {
 
   private V2Migration() { }
 
   public static final Logger SDK_V2_UPGRADE_LOG = LoggerFactory.getLogger(SDK_V2_UPGRADE_LOG_NAME);
 
-  private static final LogExactlyOnce WARN_ON_DELEGATION_TOKENS =
-      new LogExactlyOnce(SDK_V2_UPGRADE_LOG);
-
-  private static final LogExactlyOnce WARN_ON_GET_S3_CLIENT =
-      new LogExactlyOnce(SDK_V2_UPGRADE_LOG);
-
   private static final LogExactlyOnce WARN_OF_DIRECTLY_REFERENCED_CREDENTIAL_PROVIDER =
       new LogExactlyOnce(SDK_V2_UPGRADE_LOG);
 
-  private static final LogExactlyOnce WARN_OF_CUSTOM_SIGNER =
-      new LogExactlyOnce(SDK_V2_UPGRADE_LOG);
-
   private static final LogExactlyOnce WARN_OF_REQUEST_HANDLERS =
       new LogExactlyOnce(SDK_V2_UPGRADE_LOG);
 
-  private static final LogExactlyOnce WARN_ON_GET_OBJECT_METADATA =
-      new LogExactlyOnce(SDK_V2_UPGRADE_LOG);
-
   /**
-   * Warns on an AWS V1 credential provider being referenced directly.
+   * Notes an AWS V1 credential provider being referenced directly.
    * @param name name of the credential provider
    */
   public static void v1ProviderReferenced(String name) {
```

Comment:
currently this isn't used anywhere. CredentialProviderListFactory already has LOG_REMAPPED_ENTRY which will log if you're using a V1 provider I think. if we want the debug..we can call this from CredentialProviderListFactory.buildAWSProviderList

Reply:
cut it.